src/umpire/Allocator.inl

```diff
@@ -88,6 +100,29 @@ inline void* Allocator::do_named_allocate(const std::string& name, std::size_t b
   return ret;
 }
 
+inline void* Allocator::do_resource_allocate(camp::resources::Resource const& r, std::size_t bytes)
+{
+  void* ret = nullptr;
+
+  UMPIRE_ASSERT(UMPIRE_VERSION_OK());
+
+  UMPIRE_LOG(Debug, "(" << bytes << ")");
+
+  if (0 == bytes) {
+    ret = allocateNull();
+  } else {
+    ret = m_allocator->allocate_resource(r, bytes);
+  }
+
+  if (m_tracking) {
+    registerAllocation(ret, bytes, m_allocator);
+  }
+
+  umpire::event::record<umpire::event::allocate_resource>(
+      [&](auto& event) { event.size(bytes).ref((void*)m_allocator).ptr(ret).res(camp::resources::to_string(r)); });
```

Comment:
Why does the resource get stringified?

Reply:
We currently don't have a way to do anything with the resource once it is inside all the event stuff. I tried to do that (pass in the resource), but there was some issue with the default constructor being deleted or something along those lines. Doing anything further with the resource within umpire events will be the subject of another PR, if it's needed. Currently we will just print out the resource used as a string

Reply:
@davidbeckingsale - that's what I remember anyways, there was an issue with the constructor... am i remembering correctly?